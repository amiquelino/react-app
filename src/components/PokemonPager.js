import React from 'react'
import {Button, Pager} from 'react-bootstrap'
import PropTypes from 'prop-types'

const PokemonPager = (props ) => {
  return (
    <Pager>
      <Button disabled={ !props.previous? true : false} onClick={props.previousOnClick}>Anterior</Button>
        {' '}
      <Button disabled={ !props.next? true : false } onClick={props.nextOnClick}>Pr√≥xima</Button>
    </Pager>
  )
}

PokemonPager.propTypes = {
  previous: PropTypes.string,
  next: PropTypes.string,
  previousOnClick: PropTypes.func.isRequired,
  nextOnClick: PropTypes.func.isRequired,
}

export default PokemonPager
