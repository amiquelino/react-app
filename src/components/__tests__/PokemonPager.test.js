import React from 'react'
import renderer from 'react-test-renderer'
import PokemonPager from '../PokemonPager'

describe('PokemonPager', ( ) => {
  it('render correctly', ( ) => {
    const component = renderer.create(
      <PokemonPager
        next={'http://pokeapi.co/api/v2/pokemon/?offset=20'}
        previous={null}
        previousOnClick={jest.fn()}
        nextOnClick={jest.fn()}
      />
    )
    expect( component ).toMatchSnapshot( )
  })
})
