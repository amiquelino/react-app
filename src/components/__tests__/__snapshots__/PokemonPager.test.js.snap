// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPager render correctly 1`] = `
<ul
  className="pager"
>
  <button
    className="btn btn-default"
    disabled={true}
    onClick={[MockFunction]}
    onSelect={null}
    type="button"
  >
    Anterior
  </button>
   
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[MockFunction]}
    onSelect={null}
    type="button"
  >
    Pr√≥xima
  </button>
</ul>
`;
